models/issue_assignees.go

```diff
@@ -0,0 +1,232 @@
+// Copyright 2018 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package models
+
+import (
+	"fmt"
+
+	"code.gitea.io/gitea/modules/log"
+	api "code.gitea.io/sdk/gitea"
+
+	"github.com/go-xorm/xorm"
+)
+
+// IssueAssignees saves all issue assignees
+type IssueAssignees struct {
+	ID         int64 `xorm:"pk autoincr"`
+	AssigneeID int64 `xorm:"INDEX"`
+	IssueID    int64 `xorm:"INDEX"`
+}
+
+// This loads all assignees of an issue
+func (issue *Issue) loadAssignees(e Engine) (err error) {
+	var assigneeIDs []IssueAssignees
+
+	err = e.Where("issue_id = ?", issue.ID).Find(&assigneeIDs)
+	if err != nil {
+		return
+	}
+
+	for _, assignee := range assigneeIDs {
+		user, err := getUserByID(e, assignee.AssigneeID)
```

Comment:
Maybe you can use
```Go
var assigneeIDs []int64
err = e.Table("issue_assignees").Where("issue_id = ?", issue.ID).Find(&assigneeIDs)
```
and then
```Go
issue.Assignees = make([]*User, 0, len(assigneeIDs))
err = e.In("id", assigneeIDs).Find(&issue.Assignees)
```

Reply:
Great hint, thanks. I'd even go as far and do it all in one go with a `join`. How should I check if the user still exists though? (See line 35)

Reply:
@kolaente you can compare `issue.Assignees` and `assigneeIDs` on the memory for check.

Reply:
I've made it in one single join, without an extra check if the user exists or not as it is impossible to fetch nonexistent users with a join.